token/program-2022/src/offchain.rs

```diff
@@ -46,14 +61,1005 @@ where
+        let mut extra_meta_4_data = vec![]; // Mock
+        let mut extra_meta_4_lamports = 0; // Mock
+        let extra_meta_4_account_info = AccountInfo::new(
+            &extra_meta_4_pubkey,
+            false,
+            true,
+            &mut extra_meta_4_lamports,
+            &mut extra_meta_4_data,
+            &transfer_hook_program_id,
+            true, // Executable program
+            0,
+        );
+
+        let extra_meta_5_pubkey = Pubkey::find_program_address(
+            &[
+                b"prefix",
+                amount.to_le_bytes().as_ref(), // Instruction data 8..16
+                extra_meta_2_pubkey.as_ref(),
+            ],
+            &extra_meta_4_pubkey, // PDA off of the arbitrary program ID
+        )
+        .0;
+        let mut extra_meta_5_data = vec![]; // Mock
+        let mut extra_meta_5_lamports = 0; // Mock
+        let extra_meta_5_account_info = AccountInfo::new(
+            &extra_meta_5_pubkey,
+            false,
+            true,
+            &mut extra_meta_5_lamports,
+            &mut extra_meta_5_data,
+            &extra_meta_4_pubkey,
+            false,
+            0,
+        );
+
+        let extra_meta_6_pubkey = Pubkey::find_program_address(
+            &[
+                b"another_prefix",
+                amount.to_le_bytes().as_ref(), // Instruction data 8..16
+                extra_meta_2_pubkey.as_ref(),
+                extra_meta_5_pubkey.as_ref(),
+            ],
+            &extra_meta_4_pubkey, // PDA off of the arbitrary program ID
+        )
+        .0;
+        let mut extra_meta_6_data = vec![]; // Mock
+        let mut extra_meta_6_lamports = 0; // Mock
+        let extra_meta_6_account_info = AccountInfo::new(
+            &extra_meta_6_pubkey,
+            false,
+            true,
+            &mut extra_meta_6_lamports,
+            &mut extra_meta_6_data,
+            &extra_meta_4_pubkey,
+            false,
+            0,
+        );
+
+        let mut validate_state_data = MOCK_EXTRA_METAS_STATE.to_vec();
+        let mut validate_state_lamports = 0; // Mock
+        let validate_state_account_info = AccountInfo::new(
+            &validate_state_pubkey,
+            false,
+            true,
+            &mut validate_state_lamports,
+            &mut validate_state_data,
+            &transfer_hook_program_id,
+            false,
+            0,
+        );
+
+        // First use the transfer instruction builder function to add the extra
+        // account metas to the transfer instruction from offchain
+        let offchain_transfer_instruction = create_transfer_instruction_with_extra_metas(
+            &spl_token_2022_program_id,
+            &source_pubkey,
+            &mint_pubkey,
+            &destination_pubkey,
+            &authority_pubkey,
+            &[],
+            amount,
+            9,
+            mock_fetch_account_data_fn,
+        )
+        .await
+        .unwrap();
```

Comment:
Is this the only difference in this test? Since they're quite long, is it possible to add this instruction as another element to check against in the previous test and remove this one?

If not, can you create two `test_case`s for these? That will make the difference with these two tests clear

Reply:
I'll give the tests a once-over and try to shorten/tighten them up. As per my other comment above, we may not even need the first.